### PR TITLE
[RFC] Update version.c to 7.4.1219

### PR DESCRIPTION
**Pull request by Shougo**
Related issue: https://github.com/neovim/neovim/pull/3919

included patches:
 697 Already included e4975f8
1013
1149 Revert 7.4.1013

NA patches:
1002 Windows Test
1003 Travis CI
1004 Makefile
1008 Drop OS/2 support
1009 ARCHIE is already removed in neovim
1011 if_perl
1012 if_python
1014 Cygwin support
1016 Drop OS/2 support
1019 Makefile fix
1020 Windows fix
1021 Makefile fix
1022 Remove outdated description
1023 Makefile fix
1024 Windows fix
1025 Makefile fix
1026 Makefile fix
1027 if_python fix
1028 Nsis version file
1031 if_python
1033 Windows fix
1038 configure fix
1040 tee.c
1043 INSTALL_mac.txt
1044 ifdef FEAT_EVAL
1045 Travis fix
1060 old style test README fix
1062 Makefile fix
1063 Makefile fix
1065 if_python, if_lua, if_perl, if_ruby fix
1066 Windows build
1067 Makefile fix
1069 Compiler warning
1070 if_tcl
1074 GUI Compiler warning
1077 Windows build documentation
1078 Windows build
1080 Windows GUI
1082 Windows GUI
1083 GVimExt
1095 Makefile
1098 style fix
1099 crypt
1100 Cygwin Makefile
1103 filelist
1104 if_mzscheme
1106 Windows Makefile
1109 Windows
1115 Windows Makefile
1124 Windows
1125 if_perl
1127 if_perl
1128 Windows junction delete() fix
1129 if_python
1131 viminfo
1133 ARGS() macro
1134 Windows
1135 Windows
1138 GUI
1139 Windows
1144 float.h
1145 configure fix
1146 if_python3 fix
1148 Makefile fix
1155 feature remove
1158 ARGS macro
1159 ARGS macro
1162 if_mzscheme
1171 Makefile fix
1172 configure fix
1175 MinGW Cygwin build
1176 channel prototype fix
1185 if_tcl
1190 configure fix
1192 FEAT non multibyte fix
1196-1203 ARGS macro
1204 GUI Motif
1205-1208 old style function
1209 Athena support
1210 old style function
1211 old style function
1212 GUI Motif
1213-1215 old style function
1216 HAVE_STDARG_H
1218 old style function
1219 FEAT_FLOAT

### CODE REVIEW

**Comment by sethjackson** [comment]
:+1: LGTM

Also I don't believe the following are relevant:

* 1060 - old style test README fix
* 1067 - Makefile fix

**Comment by Shougo** [comment]
Thank you.  I updated it.

**Comment by Shougo** [comment]
I have updated it to 7.4.1152

**Comment by sethjackson** [comment]
* 1122 is still relevant (we still have that Makefile in `src/nvim/testdir`)
* 1128 is tricky. While we removed the code for this stuff and use `libuv` now we might have the same problem Vim does. Did not investigate.

Otherwise LGTM.

**Comment by Shougo** [comment]
Thank you!

Yes, 1122 is not NA.
I don't know 1128 is NA.  But I have removed it from NA list.

@equalsraf Could you check it?

File: src/nvim/version.c [hunk 1]
```diff
@@ -69,6 +69,103 @@ static char *features[] = {
 
 // clang-format off
 static int included_patches[] = {
+  // 1152,
+  // 1151,
+  // 1150,
+  1149,
```
Comment (oni-link) [inline src/nvim/version.c]: This patch is mentioned as NA in the commit message.

File: src/nvim/version.c [hunk 2]
```diff
@@ -79,50 +176,50 @@ static int included_patches[] = {
...
-  // 1027,
-  // 1026,
-  // 1025,
-  // 1024,
-  // 1023,
-  // 1022,
-  // 1021,
-  // 1020,
-  // 1019,
+  // 1027 NA,
```
Comment (oni-link) [inline src/nvim/version.c]: Is not mentioned in commit message. Perhaps `vim_str2nr()` parameter change could be useful.

File: src/nvim/version.c [hunk 3]
```diff
@@ -79,50 +176,50 @@ static int included_patches[] = {
...
   // 1035,
   // 1034,
-  // 1033,
+  // 1033 NA,
   1032,
-  // 1031,
+  // 1031 NA,
   // 1030,
   // 1029,
   // 1028,
```
Comment (oni-link) [inline src/nvim/version.c]: Commit message says this patch is NA.

File: src/nvim/version.c [hunk 4]
```diff
@@ -69,6 +69,103 @@ static char *features[] = {
...
+  // 1074 NA,
+  // 1073,
+  // 1072,
+  // 1071,
+  // 1070 NA,
+  // 1069 NA,
+  // 1068,
+  // 1067 NA,
+  // 1066 NA,
+  // 1065 NA,
```
Comment (oni-link) [inline src/nvim/version.c]: Is not mentioned in commit message.

**Comment by equalsraf** [comment]
> I don't know 1128 is NA. But I have removed it from NA list.

Doesn't seem to apply - I tried removing a junction with delete() and it was removed.

File: src/nvim/version.c (hunk 1 above)
Comment (Shougo) [inline src/nvim/version.c]: Because, it is revert path(no means).
I will fix the commit message.

File: src/nvim/version.c (hunk 3 above)
Comment (Shougo) [inline src/nvim/version.c]: Thank you.  Fixed.

File: src/nvim/version.c (hunk 4 above)
Comment (Shougo) [inline src/nvim/version.c]: Fixed.

**Comment by Shougo** [comment]
> Doesn't seem to apply - I tried removing a junction with delete() and it was removed.

Thank you.

**Comment by Shougo** [comment]
Updated.

File: src/nvim/version.c (hunk 3 above)
Comment (oni-link) [inline src/nvim/version.c]: It is still missing suffix `NA`.

**Comment by jusga** [comment]
697 (vim/vim@d94682ffb4fb479e99f375431f82d67b38255629) can be marked as included: https://github.com/neovim/neovim/commit/e4975f82c9704aa5d52e18e578a449268654ee37.

File: src/nvim/version.c (hunk 2 above)
Comment (oni-link) [inline src/nvim/version.c]: In the commit message this is named as `if_python fix` but it should be `bin support` if included.

File: src/nvim/version.c (hunk 3 above)
Comment (Shougo) [inline src/nvim/version.c]: Fixed.

**Comment by Shougo** [comment]
@jusga Thank you.  Marked.

**Comment by Shougo** [comment]
Updated to Vim 7.4.1219.

**Comment by oni-link** [comment]
Except four changes mentioned earlier, LGTM.

**Comment by justinmk** [comment]
Merging now with @oni-link 's latest comments.

**Comment by Shougo** [comment]
Thank you!

**Pull request closed without merging**